packages/synapse-react-client/src/components/DataGrid/columns/AutocompleteColumn.tsx

```diff
@@ -86,34 +114,29 @@ export function AutocompleteCell({
           setRowData(newVal)
         }
         // Update local input state to match the selected/created value
-        setLocalInputState(castCellValueToString(newVal))
+        // setLocalInputState(castCellValueToString(newVal))
         setTimeout(() => stopEditing({ nextRow: false }), 0)
       }}
       blurOnSelect={true}
-      onBlur={_event => {
-        // Only update on blur if the input value differs from the current rowData
-        // and no option was selected (which would have already updated via onChange)
-        if (localInputState !== castCellValueToString(rowData)) {
-          setRowData(parseFreeTextGivenJsonSchemaType(localInputState, colType))
-        }
-      }}
```

Comment:
This old `onBlur` would be invoked when an item was selected, overwriting the selection!